Library/Homebrew/cmd/update-report.rb

```diff
@@ -126,132 +127,143 @@ def output_update_report
+          hub.dump(updated_formula_report: !args.auto_update?) unless args.quiet?
+          hub.reporters.each(&:migrate_tap_migration)
+          hub.reporters.each { |r| r.migrate_formula_rename(force: args.force?, verbose: args.verbose?) }
 
-        CacheStoreDatabase.use(:descriptions) do |db|
-          DescriptionCacheStore.new(db)
-                               .update_from_report!(hub)
-        end
-        CacheStoreDatabase.use(:cask_descriptions) do |db|
-          CaskDescriptionCacheStore.new(db)
-                                   .update_from_report!(hub)
+          CacheStoreDatabase.use(:descriptions) do |db|
+            DescriptionCacheStore.new(db)
+                                 .update_from_report!(hub)
+          end
+          CacheStoreDatabase.use(:cask_descriptions) do |db|
+            CaskDescriptionCacheStore.new(db)
+                                     .update_from_report!(hub)
+          end
         end
+        puts if args.auto_update?
+      elsif !args.auto_update? && !ENV["HOMEBREW_UPDATE_FAILED"] && !ENV["HOMEBREW_MIGRATE_LINUXBREW_FORMULAE"]
+        puts "Already up-to-date." unless args.quiet?
       end
-      puts if args.auto_update?
-    elsif !args.auto_update? && !ENV["HOMEBREW_UPDATE_FAILED"] && !ENV["HOMEBREW_MIGRATE_LINUXBREW_FORMULAE"]
-      puts "Already up-to-date." unless args.quiet?
-    end
 
-    Commands.rebuild_commands_completion_list
-    link_completions_manpages_and_docs
-    Tap.each(&:link_completions_and_manpages)
+      Commands.rebuild_commands_completion_list
+      link_completions_manpages_and_docs
+      Tap.each(&:link_completions_and_manpages)
 
-    failed_fetch_dirs = ENV["HOMEBREW_MISSING_REMOTE_REF_DIRS"]&.split("\n")
-    if failed_fetch_dirs.present?
-      failed_fetch_taps = failed_fetch_dirs.map { |dir| Tap.from_path(dir) }
+      failed_fetch_dirs = ENV["HOMEBREW_MISSING_REMOTE_REF_DIRS"]&.split("\n")
+      if failed_fetch_dirs.present?
+        failed_fetch_taps = failed_fetch_dirs.map { |dir| Tap.from_path(dir) }
 
-      ofail <<~EOS
-        Some taps failed to update!
-        The following taps can not read their remote branches:
-          #{failed_fetch_taps.join("\n  ")}
-        This is happening because the remote branch was renamed or deleted.
-        Reset taps to point to the correct remote branches by running `brew tap --repair`
-      EOS
+        ofail <<~EOS
+          Some taps failed to update!
+          The following taps can not read their remote branches:
+            #{failed_fetch_taps.join("\n  ")}
+          This is happening because the remote branch was renamed or deleted.
+          Reset taps to point to the correct remote branches by running `brew tap --repair`
+        EOS
+      end
     end
 
     return if new_repository_version.blank?
 
     puts
     ohai "Homebrew was updated to version #{new_repository_version}"
```

Comment:
Does anything below this really need to be changed? I'd be okay with the minor rewording of the messages, but I don't think the logic here needs to be any different, right?

Reply:
Ah, I see. Currently, if I update from x.1.1 to x.1.3, how many revision messages do I see? I think it's only the most recent one, correct? That feels... wrong? Maybe? TBH I'm not sure.

Maybe we should just post a generic message like:

```
Homebrew was updated to version #{new_repository_version}
Changelogs for all Homebrew versions can be found at:
  https://github.com/Homebrew/brew/releases
```

If there are one or more major/minor releases in between `old_repository_version` and `new_repository_version`, then we show this message:

```
Homebrew was updated to version #{new_repository_version}
Changelogs for all Homebrew releases can be found at:
  https://github.com/Homebrew/brew/releases
More detailed release notes for major and minor releases are available on the Homebrew Blog:
  https://brew.sh/blog/3.2.0
  https://brew.sh/blog/3.3.0
  ...
```

Directing the user to https://github.com/Homebrew/brew/releases doesn't really give them any less info than https://github.com/Homebrew/brew/releases/latest, plus it allows them to scroll.

Also, this way, we still give links to the blog posts which are really the most important.

Reply:
I made the change you suggested, with the only difference is that I made message for the blog generic too.

Reply:
Although I made the change as suggested. I think it is clear to have the blog message first as below:

```
==> Homebrew was updated to version 3.5.3
More detailed release notes for major and minor releases are available on the Homebrew Blog:
  https://brew.sh/blog/
Changelogs for all Homebrew versions can be found at:
  https://github.com/Homebrew/brew/releases
```

Reply:
Reverted per https://github.com/Homebrew/brew/pull/13482#discussion_r913653140.